catalog/nsili/catalog-nsili-common/src/main/java/com/connexta/alliance/nsili/common/BqsConverter.java

```diff
@@ -0,0 +1,1218 @@
+
+    private static final String BASIC_BQS_5f =
+            "identifierMission like 'Test' and (spatialGeographicReferenceBox intersect POINT(81:45:33.2N,146:25:01.8W))";
+
+    private static final String BASIC_BQS_5g =
+            "identifierMission like 'Test' and (spatialGeographicReferenceBox within 6000 meters of POINT(46.155441760892586,81.76504326406543))";
+
+    private static final String BASIC_BQS_5h =
+            "identifierMission like 'Test' and (spatialGeographicReferenceBox beyond 6 statute miles of CIRCLE(46.155441760892586,81.76504326406543,25.6 meters))";
+
+    private static final String BQS_TEST_STRING =
+            "((identifierUUID like 'Test') or (targetNumber like 'Test') or (source like 'Test') or "
+                    + "(identifierMission like 'Test') or (category like 'Test') or "
+                    + "(decompressionTechnique like 'Test') or (identifier like 'Test') or "
+                    + "(recipient like 'Test') or (messageBody like 'Test') or "
+                    + "(messageType like 'Test') or (classification like 'Test') or (policy like 'Test') or "
+                    + "(releasability like 'Test') or (partIdentifier like 'Test') or (creator like 'Test') or "
+                    + "(encodingScheme like 'Test')) and (dateTimeModified >= '2016/03/14 06:58:31')";
+
+    private static final String TEST_BQS_SAM1 =
+            "NSIL_CARD.identifier like '%' AND (not NSIL_PRODUCT:NSIL_CARD.status = 'OBSOLETE')";
+
+    private FilterBuilder filterBuilder;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BqsConverter.class);
+
+    public BqsConverter(FilterBuilder filterBuilder) {
+        this.filterBuilder = filterBuilder;
+    }
+
+    public Filter convertBQSToDDF(Query query) {
+        String bqsQuery = query.bqs_query;
+        return convertBQSToDDF(bqsQuery);
+    }
+
+    public Filter convertBQSToDDF(String query) {
+        query = query.trim();
+
+        LOGGER.debug("Original Query: {}", query);
+
+        ANTLRInputStream inputStream = new ANTLRInputStream(query);
+        BqsLexer lex = new BqsLexer(inputStream); // transforms characters into tokens
+        CommonTokenStream tokens = new CommonTokenStream(lex); // a token stream
+        BqsParser parser = new BqsParser(tokens); // transforms tokens into parse trees
+        BqsTreeWalkerListener bqsListener = new BqsTreeWalkerListener(filterBuilder);
+
+        ParseTree tree = parser.query();
+        ParseTreeWalker.DEFAULT.walk(bqsListener, tree);
+
+        Filter filter = bqsListener.getFilter();
+        if (filter != null) {
+            LOGGER.debug("Parsed Query: {}", filter.toString());
+        } else {
+            LOGGER.debug("Unable to parse the query");
+        }
+
+        return filter;
+    }
+
+    public static void main(String[] args) {
```

Comment:
Can we get rid of main?  This looks like test code; i feel like this should be separate from the BqsConverter